File: Chapter10/R/Example1.R
# Chapter 10 - Δειγματοληπτικές Κατανομές και Κεντρικό Οριακό Θεώρημα
# Example 1: ΚΟΘ με Ομοιόμορφη Κατανομή U(0,10)
#   n = 5, 10, 30, 100 — 10000 προσομοιώσεις ανά μέγεθος

set.seed(42)
n_simulations <- 10000
sample_sizes  <- c(5, 10, 30, 100)

# Theoretical parameters for U(0,10)
mu    <- 5
sigma <- sqrt((10 - 0)^2 / 12)

# Create layout for plots
par(mfrow = c(2, 2))

for (n in sample_sizes) {
  # Generate sample means from uniform distribution
  sample_means <- replicate(n_simulations, {
    mean(runif(n, min = 0, max = 10))
  })

  se <- sigma / sqrt(n)

  # Histogram
  hist(sample_means, breaks = 50, freq = FALSE,
       col    = "lightblue", border = "black",
       main   = paste("Sample Size n =", n),
       xlab   = "Sample Mean",
       ylab   = "Density")

  # Overlay theoretical normal curve
  x <- seq(min(sample_means), max(sample_means), length.out = 200)
  y <- dnorm(x, mean = mu, sd = se)
  lines(x, y, col = "red", lwd = 2)

  # Add vertical line at population mean
  abline(v = mu, col = "darkgreen", lwd = 2, lty = 2)

  legend("topright",
         legend = c("Simulated", "Theoretical N", "Pop. Mean"),
         col    = c("lightblue", "red", "darkgreen"),
         lwd    = c(NA, 2, 2),
         lty    = c(NA, 1, 2),
         fill   = c("lightblue", NA, NA))
}

par(mfrow = c(1, 1))

File: Chapter10/Python/Example1.py
# Chapter 10 - Δειγματοληπτικές Κατανομές και Κεντρικό Οριακό Θεώρημα
# Example 1: ΚΟΘ με Ομοιόμορφη Κατανομή U(0,10)
#   n = 5, 10, 30, 100 — 10000 προσομοιώσεις ανά μέγεθος

import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import norm

np.random.seed(42)
n_simulations = 10000
sample_sizes  = [5, 10, 30, 100]

# Theoretical parameters for U(0,10)
mu    = 5
sigma = np.sqrt((10 - 0)**2 / 12)

fig, axes = plt.subplots(2, 2, figsize=(14, 10))
axes = axes.ravel()

for idx, n in enumerate(sample_sizes):
    # Generate sample means from uniform distribution
    sample_means = [np.mean(np.random.uniform(0, 10, n))
                    for _ in range(n_simulations)]

    se = sigma / np.sqrt(n)

    # Histogram
    axes[idx].hist(sample_means, bins=50, density=True,
                   alpha=0.6, color='blue', edgecolor='black',
                   label='Simulated')

    # Overlay theoretical normal curve
    x = np.linspace(min(sample_means), max(sample_means), 200)
    y = norm.pdf(x, loc=mu, scale=se)
    axes[idx].plot(x, y, 'r-', linewidth=2, label='Theoretical N')

    # Add vertical line at population mean
    axes[idx].axvline(mu, color='green', linestyle='--',
                      linewidth=2, label='Pop. Mean')

    axes[idx].set_xlabel('Sample Mean', fontsize=11)
    axes[idx].set_ylabel('Density', fontsize=11)
    axes[idx].set_title(f'Sample Size n = {n}', fontsize=12)
    axes[idx].legend(fontsize=9)
    axes[idx].grid(alpha=0.3)

plt.suptitle('CLT Demonstration — Uniform Distribution U(0,10)',
             fontsize=14, y=1.01)
plt.tight_layout()
plt.show()
